### PR TITLE
feat(linter): default config path for linter

### PR DESCRIPTION
**Pull request by tapanprakasht**
Implemented read default configuration from the root folder when config is not passed via the parameter.  

https://github.com/oxc-project/oxc/issues/3398

### CODE REVIEW

**Comment by Dunqing** [comment]
> @Dunqing Need your input for adding the test cases. Here this https://github.com/oxc-project/oxc/tree/main/apps/oxlint folder act as the root folder for running the test cases. If I add oxlintrc.json file in the root folder other test cases will fail. Should I add test fixture in the root file ? https://github.com/oxc-project/oxc/tree/main/apps/oxlint

Can we change `cwd`? I think we should add a fixture to test it.

**Comment by Dunqing** [comment]
If the default config file doesn't exist will panic. We should only check file weather exists if provide a custom config path
https://github.com/oxc-project/oxc/blob/dd42f279f77b1fc53f22ed1de29d8ba52548bd84/crates/oxc_linter/src/config/mod.rs#L64-L68

**Comment by tapanprakasht** [comment]
@Dunqing I have added some test cases. Please check.

File: apps/oxlint/oxlintrc.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
[the remark this answers is not in the document]
Comment (Dunqing) [reply apps/oxlint/oxlintrc.json:1]: Yes, add another test without `oxlintrc.json`

[the remark this answers is not in the document]
Comment (tapanprakasht) [reply apps/oxlint/oxlintrc.json:1]: Since we want to verify default configuration with `oxlintrc.json`, how to a add test without having this file?

**Comment by Boshen** [comment]
Thank you for your contribution. There was a miscommunication about this feature.

Apparently ESLint and most tools don't do this type of default. It's a confusing behavior. So I'm going to close this feature as not planned.

**Pull request closed without merging**